drgn_tools/task.py

```diff
@@ -203,6 +336,30 @@ def show_tasks_last_runtime(tasks: Iterable[Object]) -> None:
     print_table(rows)
 
 
+def show_taskinfo(tasks: Iterable[Object]) -> None:
+    """
+    Display task information.
+    """
+    rows = [["PID", "PPID", "CPU", "TASK", "ST", "%MEM", "VSZ", "RSS", "COMM"]]
+    tasks = list(tasks)
+    tasks.sort(key=get_pid)
+    for t in tasks:
+        rows.append(
+            [
+                str(get_pid(t)),
+                str(get_ppid(t)),
+                str(task_cpu(t)),
+                hex(t.value_()),
+                task_state_to_char(t),
+                str("%.1f" % get_pct_mem(t)),
+                str(get_vmem(t)),
+                str(get_rss(t)),
+                get_command(t),
+            ]
+        )
```

Comment:
In order to gain the benefit of caching with `get_task_rss()`, you need to create the cache here and pass it in to each function call. I'd suggest also getting rid of `get_rss()` and `get_pct_mem()` since they both need this data, and it's simpler to just fetch it once and compute it all here.
```suggestion
def show_taskinfo(prog: Program, tasks: Iterable[Object]) -> None:
    """
    Display task information.
    """
    rows = [["PID", "PPID", "CPU", "TASK", "ST", "%MEM", "VSZ", "RSS", "COMM"]]
    tasks = list(tasks)
    tasks.sort(key=get_pid)
    page_size = int(prog["PAGE_SIZE"])
    total_mem = int(totalram_pages(prog))
    rss_cache = {}
    for t in tasks:
        task_rss = get_task_rss(t, rss_cache)
        rss_kb = task_rss.total * page_size // ByteToKB
        pct_mem = task_rss.total * 100 / pct_mem
        rows.append(
            [
                str(get_pid(t)),
                str(get_ppid(t)),
                str(task_cpu(t)),
                hex(t.value_()),
                task_state_to_char(t),
                str("%.1f" % pct_mem),
                str(get_vmem(t)),
                str(rss_kb),
                get_command(t),
            ]
        )
```